Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/cluster/ClusterDetail.java

```diff
@@ -370,56 +371,60 @@ private IHDIStorageAccount getDefaultStorageAccount(Map<String, String> coresite
+        try {
+            switch (pathInfo.storageType) {
+                case ADLS:
+                    return new ADLSStorageAccount(this, true, clusterIdentity, URI.create(defaultStorageRootPath));
 
             case BLOB:
                 String storageAccountName = pathInfo.path.getHost();
                 if (StringUtils.isBlank(storageAccountName)) {
                     throw new HDIException("Failed to get default storage account name");
                 }
 
-                String defaultContainerName = pathInfo.path.getUserInfo();
-                String defaultStorageAccountKey = StorageAccountKeyPrefix + storageAccountName;
-                String storageAccountKey = null;
-                if (coresiteMap.containsKey(defaultStorageAccountKey)) {
-                    storageAccountKey = coresiteMap.get(defaultStorageAccountKey);
-                }
+                    String defaultContainerName = pathInfo.path.getUserInfo();
+                    String defaultStorageAccountKey = StorageAccountKeyPrefix + storageAccountName;
+                    String storageAccountKey = null;
+                    if (coresiteMap.containsKey(defaultStorageAccountKey)) {
+                        storageAccountKey = coresiteMap.get(defaultStorageAccountKey);
+                    }
 
-                if (storageAccountKey == null) {
-                   return null;
-                }
+                    if (storageAccountKey == null) {
+                        return null;
+                    }
 
-                return new HDStorageAccount(this, storageAccountName, storageAccountKey, true, defaultContainerName);
+                    return new HDStorageAccount(this, storageAccountName, storageAccountKey, true, defaultContainerName);
 
-            case ADLSGen2:
-                String accountName = pathInfo.path.getHost();
-                if (StringUtils.isBlank(accountName)) {
-                    throw new HDIException("Failed to get default storage account name");
-                }
+                case ADLSGen2:
+                    String accountName = pathInfo.path.getHost();
+                    if (StringUtils.isBlank(accountName)) {
+                        throw new HDIException("Failed to get default storage account name");
+                    }
 
-                String fileSystem = pathInfo.path.getUserInfo();
-                return new ADLSGen2StorageAccount(this, accountName, null, true, fileSystem, pathInfo.path.getScheme());
+                    String fileSystem = pathInfo.path.getUserInfo();
+                    return new ADLSGen2StorageAccount(this, accountName, null, true, fileSystem, pathInfo.path.getScheme());
 
-            default:
-                return null;
+                default:
+                    return null;
+            }
+        } catch (HDIException ex) {
+            throw new HDIException(String.format("Encounter exception when getting storage configuration for root path %s with type %s",
```

Comment:
Can it be wrapped as a classified Exception?

Reply:
No , all the methods call this function will catch the exception and return empty or null and this exception can't reach to the classified exception stage.